webui/react/src/components/DeterminedAuth.tsx

```diff
@@ -30,9 +30,10 @@ const DeterminedAuth: React.FC = () => {
       const user = await getCurrentUser({});
       setAuth({ type: Auth.ActionType.Set, value: { isAuthenticated: true, user } });
     } catch (e) {
-      const actionMsg = isBadCredentials ? 'check your username and password.' : 'retry.';
+      const isBadCredentialsSync = isLoginFailure(e);
+      setIsBadCredentials(isBadCredentialsSync); // this is not a sync operation
       setShowSpinner({ type: FullPageSpinner.ActionType.Hide });
-      setIsBadCredentials(isLoginFailure(e));
+      const actionMsg = isBadCredentialsSync ? 'check your username and password.' : 'retry.';
```

Comment:
chore: update the `isBadCredentials` in `handleError` below to use `isBadCredentialsSync`